includes/api/wc-blocks/class-wc-rest-blocks-products-controller.php

```diff
@@ -0,0 +1,379 @@
+		$attr_operator      = $request->get_param( 'attr_operator' );
+		$catalog_visibility = $request->get_param( 'catalog_visibility' );
+
+		$ordering_args   = WC()->query->get_catalog_ordering_args( $orderby, $order );
+		$args['orderby'] = $ordering_args['orderby'];
+		$args['order']   = $ordering_args['order'];
+		if ( $ordering_args['meta_key'] ) {
+			$args['meta_key'] = $ordering_args['meta_key']; // WPCS: slow query ok.
+		}
+
+		if ( $cat_operator && isset( $args['tax_query'] ) ) {
+			foreach ( $args['tax_query'] as $i => $tax_query ) {
+				if ( 'product_cat' === $tax_query['taxonomy'] ) {
+					$args['tax_query'][ $i ]['operator']         = $cat_operator;
+					$args['tax_query'][ $i ]['include_children'] = 'AND' === $cat_operator ? false : true;
+				}
+			}
+		}
+
+		if ( $attr_operator && isset( $args['tax_query'] ) ) {
+			foreach ( $args['tax_query'] as $i => $tax_query ) {
+				if ( in_array( $tax_query['taxonomy'], wc_get_attribute_taxonomy_names(), true ) ) {
+					$args['tax_query'][ $i ]['operator'] = $attr_operator;
+				}
+			}
+		}
+
+		if ( in_array( $catalog_visibility, array_keys( wc_get_product_visibility_options() ), true ) ) {
+			$exclude_from_catalog = 'search' === $catalog_visibility ? '' : 'exclude-from-catalog';
+			$exclude_from_search  = 'catalog' === $catalog_visibility ? '' : 'exclude-from-search';
+
+			$args['tax_query'][] = array(
+				'taxonomy' => 'product_visibility',
+				'field'    => 'name',
+				'terms'    => array( $exclude_from_catalog, $exclude_from_search ),
+				'operator' => 'hidden' === $catalog_visibility ? 'AND' : 'NOT IN',
+			);
+		}
+
+		return $args;
+	}
+
+	/**
+	 * Get product data.
+	 *
+	 * @param WC_Product $product Product instance.
+	 * @param string     $context Request context.
+	 *                            Options: 'view' and 'edit'.
+	 * @return array
+	 */
+	protected function get_product_data( $product, $context = 'view' ) {
+		$raw_data = parent::get_product_data( $product, $context );
+		$data     = array();
+
+		$data['id']                = $raw_data['id'];
+		$data['name']              = $raw_data['name'];
+		$data['permalink']         = $raw_data['permalink'];
+		$data['sku']               = $raw_data['sku'];
+		$data['description']       = $raw_data['description'];
+		$data['short_description'] = $raw_data['short_description'];
+		$data['price']             = $raw_data['price'];
+		$data['price_html']        = $raw_data['price_html'];
+		$data['images']            = $raw_data['images'];
+
+		return $data;
+	}
+
+	/**
+	 * Update the collection params.
+	 *
+	 * Adds new options for 'orderby', and new parameters 'cat_operator', 'attr_operator'.
+	 *
+	 * @return array
+	 */
+	public function get_collection_params() {
+		$params                       = parent::get_collection_params();
+		$params['orderby']['enum']    = array_merge( $params['orderby']['enum'], array( 'price', 'popularity', 'rating', 'menu_order' ) );
+		$params['cat_operator']       = array(
```

Comment:
We generally don't use short names on REST API, also this comes from our shortcode, I don't like to expose it here... So last move way from short names, let's change to `category_operator` and `attribute_operator`.